site/docs/architecture/the-cli-crate.md

```diff
@@ -0,0 +1,64 @@
+---
+sidebar_position: 4
+---
+
+# The `cli` crate
+
+The `cli` crate contains binaries for managing different aspects of a Gerust project, mainly generating new files of various types, as well as executing database commands such as running migrations, etc. The `cli` command is part of the Gerust project for 2 reasons: a) to avoid an external dependency of the application on the `gerust` crate and b) to make the project fully functional with only a working Rust toolchain – more on both goals in the [architecture docs](../architecture).
+
+A developer working on a Gerust project would typically not have to make changes to anything inside the `cli` crate directly, although in a future version we might add a mechanism for defining custom tasks inside the `cli` crate, similar to e.g. [how Ruby on Rails does it](https://guides.rubyonrails.org/command_line.html#custom-rake-tasks).
+
+Gerust creates a workspace that is configured so that both binaries can conveniently be invoked as `cargo generate` and `cargo db` instead of the more convoluted `run --package my-app-cli --bin generate` and `run --package my-app-cli --bin db`.
+
+## The `generate` binary
```

Comment:
This part could do with (reference to) a description of what files each generate subcommand generates and alters, and how that affect whether or not you can alter (parts of) some files

Reply:
Simply knowing whether altering some modules yourself will mess up code generation is nice to know at this point

Reply:
That makes sense 👍 Can you prepare a PR adding that?

Reply:
@hdoordt